### PR TITLE
Fix cases where the input string represents 0

### PR DESCRIPTION
**Pull request by michaelhidalgo**
Fixing Issue https://github.com/unshiftio/millisecond/issues/5

### CODE REVIEW

**Comment by lpinca** [comment]
Hi,
thank you for spotting this.

The second capturing group of the regex can be `undefined` only when using a string like `'123'` or `'1.23'` or `'.123'` but when this happens we shouldn't even run `exec()` as these cases should be handled [here](https://github.com/unshiftio/millisecond/blob/994ef3ad89cb4b1e905b3d2a5c8b8938b786496b/index.js#L47)

I would only refactor [these lines](https://github.com/unshiftio/millisecond/blob/994ef3ad89cb4b1e905b3d2a5c8b8938b786496b/index.js#L45-L47) to something like:

```js
if ('number' === type) return ms;
if ('string' !== type) return 0;
if (!isNaN(+ms)) return +ms;
```

to cover all cases (including `'00'` and `'0.0'`).

**Comment by michaelhidalgo** [comment]
Yeah, that makes sense,that's a better fix.

Thank you!. I'm going to close this pull request then.

**Comment by lpinca** [comment]
You can just change your commit and I'll merge it. I reopen this.

**Comment by lpinca** [comment]
If you don't have the time to work on this, let me know :)

**Comment by michaelhidalgo** [comment]
Ok cool, I'll send the fix this afternoon!.

Thank you.

**Comment by michaelhidalgo** [comment]
Hi @lpinca , could you please review the fix!. Really nice way to fix it.

File: test.js [hunk 1]
```diff
@@ -22,7 +22,9 @@ describe('millisecond', function () {
   });
 
   it('should return 0 if invalid', function () {
-    assume(ms('Hello mom')).to.equal(0);
+    var s = ms('[/\]')
+    console.log(s)
```
Comment (3rd-Eden) [inline test.js]: I would suggest removing this debug code, keeping the original assume and add your commented out assume statement.

**Comment by 3rd-Eden** [comment]
One small comment about a clean up but LGTM otherwise.

File: test.js [hunk 2]
```diff
@@ -94,4 +96,10 @@ describe('millisecond', function () {
   it('should work with numbers starting with .', function () {
     assume(ms('.5ms')).to.equal(0.5);
   });
+
+  it('should check missing time fractions)', function(){
```
Comment (lpinca) [inline test.js]: 'should return 0 if the input string represents 0'

File: test.js (hunk 1 above)
Comment (lpinca) [inline test.js]: The comment should be removed too or it will throw (it does not resolve to 0).

**Comment by lpinca** [comment]
@michaelhidalgo squash the two commits, follow our comments and we are ready to go.
Thank you.

**Comment by lpinca** [comment]
There were some more style nits, so instead of pestering you I grabbed your commit and edited it.
Landed in 77e92c4.

Thank you very much!

**Comment by michaelhidalgo** [comment]
Thanks guys, thanks for renaming the test, you know what they say : 

"There is also a variation on this that says there are two hard things in computer science: cache invalidation, naming things, and off-by-one errors." :)

**Pull request closed without merging**